### DISCUSSION TITLE
I am trying to run a basic example of roswrapper and airsim. drone takeoff!  ( ros2 foxy )

### DISCUSSION THREAD

**Discussion by EXPX3** [microsoft/AirSim#4879] [Support Q&A]
**I am trying to run a basic example of roswrapper and airsim. drone takeoff!**
**Below is my setup.**
- on ubuntu 
- airsim and UR(UnReal) in docker container by runing the command ./run_airsim_image_binary.sh airsim_binary:11.3.1-devel-ubuntu18.04 Blocks/LinuxNoEditor/Blocks.sh -windowed -ResX=1080 -ResY=720'
- roswrapper in docker container by running the command docker run --rm -it --network=test_network --name ros2wrapper -v <path to airsim in host>:/home/testuser/AirSim airsim-ros2:latest bash
- until this point i am able to run both docker containers and start blocks environment via airsim related container and run airsim_node via roswrapper Container, i have made sure these both docker containers communicate via a test_network i created. i can ping from one container to another.

**The Problem !**
when i run ros2 service list i dont see the takeoff, land and other publishers and subscribers listed in the doccumentation page [here](https://microsoft.github.io/AirSim/airsim_ros_pkgs/)

Can you please tell me if i am missing any crusial step inbetween? and how to proceed from here?

Many Thanky in advance!!

**Comment by EXPX3** [comment]
Hi All, when i looked at the output of the launching the airsim_node via the launch file, i noticed that the airsim_node is named as [airsim_node-2], this number means the number of similar nodes that are running. Ideally we need only one airsim_node running. so instead of running the launch file i just launched the aursim_node using ros2 run airsim_ros_pkgs airsim_node command, and the out put is
waiting for the connection xxxx

THis means that the roswrapper is not connected to the airsim simulation environment or?
how should i make sure that they are connected? how did you guys manage to set it up?, in the doccumentation i did not come across any extra step to connect the roswrapper docker and airsim from binary docker.
Any pointers or help is welcome please!

Thanks in advance.

**Reply by EXPX3** [reply]
@samuelctabor @alonfaraj @zimmy87 @jmackay2 @rajat2004 @Vu1tUrR @SJ824 @xxEoD2242
@leoborgnino @3zuli @ivannson
Please also tag anyone that can help! I just want to do a basic setup with airsim, ros2 
Would appreciate if you can throw some light here. Thankyou.

**Reply by ghost** [reply]
Hi @Robotgir I am running it on windows and using ros 1, I don't know whether I can really help you with it, because I have only tested the examples given in the documentation, and haven't moved forward yet

**Reply by EXPX3** [reply]
Hi @Vu1tUrR , thanks for sharing your setup details.

Airsim and ros2 wrapper on docker setup is what I am looking for. Let's see if other members can throw some light here.

**Comment by EXPX3** [comment]
the issue is solved it was a docker connection issue and not airsim relate!d
